src/TempleLineOfCredit.sol

```diff
@@ -98,125 +127,177 @@ contract TempleLineOfCredit is Ownable, Operators {
+     * @param collateralAmount is the amount to deposit
      */
-    function postCollateral(uint256 amount) external {
-        if (amount == 0) revert InvalidAmount(amount);
-        positions[msg.sender].collateralAmount += amount;
+    function postCollateral(uint256 collateralAmount) external {
+        if (collateralAmount == 0) revert InvalidAmount(collateralAmount);
+        positions[msg.sender].collateralAmount += collateralAmount;
         collateralToken.safeTransferFrom(
             msg.sender,
             address(this),
-            amount
+            collateralAmount 
         );
-        emit PostCollateral(msg.sender, amount);
+        emit PostCollateral(msg.sender, collateralAmount);
     }
 
-    function borrow(uint256 borrowAmount) external {
-        if (borrowAmount == 0) revert InvalidAmount(borrowAmount);
+    function borrow(address[] memory tokens, uint256[] memory borrowAmounts) external {
 
-        uint256 debtAmount =  positions[msg.sender].debtAmount;
-        if (borrowAmount > debtBalance) {
-            revert InsufficentDebtToken(debtBalance, borrowAmount);
-        }
-        if (debtAmount != 0) {
-            debtAmount = getTotalDebtAmount(msg.sender);
-        }
+        for (uint256 i =0; i < tokens.length; i++) {
+            
+            address debtToken = tokens[i];
+            uint256 borrowAmount = borrowAmounts[i];
 
-        uint256 borrowCapacity = _maxBorrowCapacity(positions[msg.sender].collateralAmount) - debtAmount;
-        debtAmount += borrowAmount;
+            if (borrowAmount == 0) revert InvalidAmount(borrowAmount);
 
-        if (debtAmount > borrowCapacity) {
-            revert InsufficentCollateral(borrowCapacity, debtAmount);
-        }
+            DebtToken memory debtTokenInfo = debtTokens[debtToken];
+            if (!debtTokenInfo.isAllowed) revert Unsupported(debtToken); 
 
-        positions[msg.sender].debtAmount = debtAmount;
-        positions[msg.sender].createdAt = block.timestamp;
-         
-        debtBalance -= borrowAmount;
-        debtToken.safeTransfer(
-            msg.sender,
-            borrowAmount 
-        );
-        emit Borrow(msg.sender, borrowAmount);
+            uint256 debtAmount =  positions[msg.sender].tokenPosition[debtToken].debtAmount;
+            if (debtAmount != 0) {
+                debtAmount = getTotalDebtAmount(debtToken, msg.sender);
+            }
+
+            uint256 borrowCapacity = _maxBorrowCapacity(positions[msg.sender].collateralAmount, debtTokenInfo.tokenPrice, debtTokenInfo.minCollateralizationRatio) - debtAmount;
+            debtAmount += borrowAmount;
+
+            if (debtAmount > borrowCapacity) {
+                revert InsufficentCollateral(debtToken, borrowCapacity, debtAmount);
+            }
+
+            TokenPosition storage userPosition = positions[msg.sender].tokenPosition[debtToken];
+
+            userPosition.debtAmount = debtAmount;
+            userPosition.createdAt = block.timestamp;
```

Comment:
The timestamp changes over time, so it's not really `createdAt`. Perhaps call `lastUpdatedAt`?